.github/workflows/build.yml

```diff
@@ -13,99 +13,115 @@ env:
   CALCROM_DISCORD_WEBHOOK_USERNAME: OK
   CALCROM_WEBHOOK_URL: ${{ secrets.WEBHOOKURL }}
-  COMPARE: 1
 
 jobs:
-  build:
+  build-heartgold:
     runs-on: ubuntu-22.04
 
     steps:
-      - name: Update and Install Software
-        run: |
-          sudo apt update
-          sudo ACCEPT_EULA=Y apt -y --fix-missing --allow-downgrades upgrade
-          sudo apt -y --allow-downgrades install g++-10-multilib linux-libc-dev binutils-arm-none-eabi p7zip-full pkg-config libpugixml-dev ppa-purge
-          sudo ppa-purge -y ppa:ubuntu-toolchain-r/test
-          sudo dpkg --add-architecture i386
-          sudo mkdir -pm755 /etc/apt/keyrings
-          sudo wget -O /etc/apt/keyrings/winehq-archive.key https://dl.winehq.org/wine-builds/winehq.key
-          sudo add-apt-repository ppa:cybermax-dexter/sdl2-backport
-          sudo wget -NP /etc/apt/sources.list.d/ https://dl.winehq.org/wine-builds/ubuntu/dists/$(lsb_release -cs)/winehq-$(lsb_release -cs).sources
-          sudo apt update
-          sudo apt -y --allow-downgrades install --install-recommends winehq-stable
-
       - name: Checkout Repo
         uses: actions/checkout@v4
 
-      - name: Setup Repo
-        run: |
-          mkdir -p ~/download
-          cd ~/download
-          wget https://github.com/pret/pokeheartgold/raw/workflows/assets/mwccarm.zip
-          wget https://github.com/pret/pokeheartgold/raw/workflows/assets/NitroSDK-3_2-060901.7z
-          unzip mwccarm.zip
-          mv mwccarm $GITHUB_WORKSPACE/tools
-          7z x NitroSDK-3_2-060901.7z
-          mv NitroSDK-3_2-060901/tools/bin $GITHUB_WORKSPACE/tools
-          mv NitroSDK-3_2-060901/include/nitro/specfiles/ARM7-TS.lcf.template $GITHUB_WORKSPACE/sub/
-          mv NitroSDK-3_2-060901/include/nitro/specfiles/ARM9-TS.lcf.template $GITHUB_WORKSPACE/
-          mv NitroSDK-3_2-060901/include/nitro/specfiles/mwldarm.response.template $GITHUB_WORKSPACE/
-        working-directory: ~
-
-      - name: Build HeartGold
+      - name: Build ROM
         env:
           GAME_VERSION: HEARTGOLD
           GAME_LANGUAGE: ENGLISH
           GAME_REVISION: 0
-        run: make -j${nproc}
+          NONMATCHING: 0
+          COMPARE: 1
+          BUILD_OUT: build/heartgold.us/
+        uses: ./.github/workflows/build-rom
+
+  build-soulsilver:
+    runs-on: ubuntu-22.04
 
-      - name: Build SoulSilver
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v4
+
+      - name: Build ROM
         env:
           GAME_VERSION: SOULSILVER
           GAME_LANGUAGE: ENGLISH
           GAME_REVISION: 0
-        run: make -j${nproc}
+          NONMATCHING: 0
+          COMPARE: 1
+          BUILD_OUT: build/soulsilver.us/
+        uses: ./.github/workflows/build-rom
+
+  build-nonmatching:
+    runs-on: ubuntu-22.04
+
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v4
+
+      - name: Build ROM
+        env:
+          GAME_VERSION: HEARTGOLD
```

Comment:
this only builds non-matching HG? not SS?

Reply:
We can add SS -- it wouldn't cost any more time. But the cases where it adds coverage seemed pretty narrow. 

I think you'd only get a build error if you don't define a SS-specific data variable. I could only find one in the game: `ov74_0223C968`

Reply:
please do add it, it could be useful to have I guess